test/testdrive/kafka-exactly-once-sink.td

```diff
@@ -0,0 +1,163 @@
+
+> CREATE MATERIALIZED VIEW input_kafka_no_byo_mview_view AS SELECT * FROM input_kafka_no_byo_mview;
+
+> CREATE MATERIALIZED VIEW input_table_mview AS SELECT a + 2 AS a , b + 10 AS b from input_table;
+
+> CREATE VIEW input_values_view AS VALUES (1), (2), (3);
+
+> CREATE MATERIALIZED VIEW input_values_mview AS VALUES (1), (2), (3);
+
+> CREATE VIEW input_kafka_no_byo_join_view AS SELECT * FROM input_kafka_byo, input_kafka_no_byo;
+
+> CREATE MATERIALIZED VIEW input_kafka_no_byo_join_mview AS SELECT * FROM input_kafka_no_byo, input_kafka_byo;
+
+> CREATE MATERIALIZED VIEW input_kafka_no_byo_scalar_subquery AS SELECT (SELECT a FROM input_kafka_no_byo LIMIT 1) FROM input_kafka_byo;
+
+> CREATE MATERIALIZED VIEW input_kafka_no_byo_derived_table AS SELECT * FROM ( SELECT * FROM input_kafka_no_byo ) AS a1;
+
+$ file-append path=static.csv
+city,state,zip
+Rochester,NY,14618
+New York,NY,10004
+"bad,place""",CA,92679
+
+> CREATE SOURCE input_csv
+  FROM FILE '${testdrive.temp-dir}/static.csv'
+  FORMAT CSV WITH 3 COLUMNS
+
+> CREATE SINK output1 FROM input_kafka_byo
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'output-view-${testdrive.seed}'
+  WITH (exactly_once=true, consistency=true)
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+
+! CREATE SINK output2 FROM input_kafka_no_byo
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'output-view-${testdrive.seed}'
+  WITH (exactly_once=true, consistency=true)
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+all input sources of a consistent Kafka sink must be replayable, materialize.public.input_kafka_no_byo is not
+
+! CREATE SINK output3 FROM input_table
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'output-view-${testdrive.seed}'
+  WITH (exactly_once=true, consistency=true)
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+all inputs of a consistent Kafka sink must be sources, materialize.public.input_table is not
+
+> CREATE SINK output4 FROM input_kafka_byo_mview
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'output-view-${testdrive.seed}'
+  WITH (exactly_once=true, consistency=true)
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+
+> CREATE SINK output4_view FROM input_kafka_byo_mview_view
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'output-view-${testdrive.seed}'
+  WITH (exactly_once=true, consistency=true)
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+
+! CREATE SINK output5 FROM input_kafka_no_byo_mview
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'output-view-${testdrive.seed}'
+  WITH (exactly_once=true, consistency=true)
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+all input sources of a consistent Kafka sink must be replayable, materialize.public.input_kafka_no_byo is not
+
+> CREATE SINK output5_view FROM input_kafka_no_byo_mview_view
```

Comment:
i believe the check should be recursive. Creating a view over a non-replayable source should not allow you to create a consistent sink.

Reply:
That's a very good catch! I was (naively) assuming that `CatalogItem.uses()` returns all transitive dependencies. I now changed `get_root_dependencies()` to properly recurse into transitive dependencies. PTAL.

Reply:
The test now shows that the correct error is returned. Thanks.